src/steamship/agents/memory/chathistory.py

```diff
@@ -36,9 +43,35 @@ def _get_existing_file(client: Client, context_keys: Dict[str, str]) -> Optional
                 "Multiple ChatHistory objects have been created in this workspace with these memory keys."
             )
 
+    @staticmethod
+    def _get_index_handle_from_file(file: File) -> str:
+        for tag in file.tags:
+            if tag.kind == TagKind.CHAT and tag.name == ChatTag.INDEX_HANDLE:
+                return tag.value[TagValueKey.STRING_VALUE]
+        raise SteamshipError(f"Could not find index handle on file with id {file.id}")
+
+    @staticmethod
+    def _get_embedding_index(client: Steamship, index_handle: str) -> EmbeddingIndexPluginInstance:
```

Comment:
Aside: variations of this method are in a few places in the codebase (I've added variations of this to a few spots in the Tools & Mixins). At some point we may want to extract it into a util method.. or maybe even a static method on EmbeddingIndexPluginIndex itself.. like `get_default` or something